### PR TITLE
feature(core): Add scoop-shim helper

### PR DESCRIPTION
**Pull request by r15ch13**
This adds support for the extracted [Shim](https://github.com/ScoopInstaller/Shim/).exe

Related: https://github.com/lukesampson/scoop/pull/3066

### CODE REVIEW

File: lib/core.ps1 [hunk 1]
```diff
@@ -427,8 +428,11 @@ function shim($path, $global, $name, $arg) {
     }
 
     if($path -match '\.(exe|com)$') {
+        if(!(Test-HelperInstalled -Helper Shim)) {
+            scoop install 'main/scoop-shim'
```
Comment (Ash258) [inline lib/core.ps1]: Manifest is missing binary. Is it intentional?

Comment (r15ch13) [reply lib/core.ps1]: Yes, what would be the purpose of shimming the shim? :smile:

Comment (Ash258) [reply lib/core.ps1]: You can use it then from anywhere and shim anything. Nice for users who want to shim custom executables.

Small workaround for #3303

**Comment by r15ch13** [comment]
The robocopy tests fail because they require a shimmed file.
I also don't know of the current position in code is a good place for the installation.
It should probably be installed on update and the first scoop installation.

File: lib/core.ps1 (hunk 1 above)
Comment (r15ch13) [reply lib/core.ps1]: I would like to avoid recursions :grin:

It could just be copied from `~/scoop/apps/scoop-shim/current/shim.exe`

**Comment by rashil2000** [comment]
Closing in favour of #4736

**Pull request closed without merging**